File: java/src/main/java/com/sitepoint/Greeter.java
package com.sitepoint;

public class Greeter {
    private String name;

    public Greeter(String name) {
        this.name = name;
    }

    public Greeter() {
        this("World");
    }

    public void sayHello() {
        String greeting = "Hello, %s!\n";
        System.out.printf(greeting, this.name);
    }

}

File: cpp/src/Greeter.cpp
#ifndef _GREETER_
#define _GREETER_

#include <iostream>

class Greeter {
    private:
    std::string name;
    
    public:
    Greeter(std::string name) {
        this->name = name;
    }
    
    Greeter() {
        this->name = "World";
    }
    
    void greet() {
        std::cout<<"Hello, "<<this->name<<"!"<<std::endl;
    }
    
};

#endif